File: Leetcode_Solutions/Striver SDE Sheet/Majority element II.cpp
#include <bits/stdc++.h> 
vector<int> majorityElementII(vector<int> &arr)
{
    // Write your code here.
    map<int,int> mp;
    vector<int> vec;
    int size = arr.size()/3;
    for(int i : arr)
    {
        if(mp[i]) mp[i]+=1;
        else mp[i] = 1;
    }
    
    int mp_size = mp.size();
    for(auto data : mp)
    {
        if (data.second > size) vec.push_back(data.first);      
    }
    return vec;
}

File: Leetcode_Solutions/Striver SDE Sheet/Majority element II.py
from os import *
from sys import *
from collections import *
from math import *

def majorityElementII(arr):
	# Write your code here.
    dict1 = dict()
    for i in arr:
        if i in dict1:
            dict1[i] += 1
        else:
            dict1[i] = 1
    val = len(arr)/3
    final = []
    for key, value in dict1.items():
        if value > val:
            final.append(key)
            
    return final
    
    
